### PR TITLE
feat(iuod27)Upgrade the volumes of application when all replica pods are running

### PR DESCRIPTION
**Pull request by amitbhatt818**
**What this PR does / why we need it**:
- Upgrade the volumes of application when all replica pods are running
**_Details:_**

**_Steps involved in openebs update in cluster:_**

- **Installing openebes**: 
  - Check whether all the components are in running state or not.
  - Create PVC which will use `storageClass: openebs-jiva-default.`
  - Deploy an application busybox which will use the above pvc.

- **Upgrading Openebs**
  -  Upgrade openebs control plane components first.
  -  Now upgrade the data plane components
  -  Check OpenEBS control plane pods state after control plane upgrade
  - Upgrade application volume
  - Check application volume health status for all the replicas.

**Folder Added:** 
-/oep/litmus/director/openebs-upgrade-jiva

Signed-off-by: Amit Bhatt <amitbhatt818@gmail.com>

### CODE REVIEW

File: litmus/director/openebs-upgrade-jiva/busybox_app.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      app: bb
+  template:
+    metadata:
+      labels:
+        app: bb
+    spec:
+      containers:
+      - name: app-busybox
+        imagePullPolicy: IfNotPresent
+        image: busybox
```
Comment (AmitKumarDas) [inline litmus/director/openebs-upgrade-jiva/busybox_app.yaml]: please check if busybox can be considered as a valid application?
cc @ajeshbaby

File: litmus/director/openebs-upgrade-jiva/test.yml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+    ## Getting the password.stdout     
+    - name: Get password
+      shell: cat /etc/secret-volume/password
+      register: password
+    
+    - block:
+        ## Install openebs using operator yaml
+        - name: Install openebs using operator yaml
+          shell: kubectl apply -f https://openebs.github.io/charts/openebs-operator-{{ openebs_current_version }}.yaml
```
Comment (AmitKumarDas) [inline litmus/director/openebs-upgrade-jiva/test.yml]: are we not supposed to use director to install?

File: litmus/director/openebs-upgrade-jiva/busybox_app.yaml (hunk 1 above)
Comment (amitbhatt818) [reply litmus/director/openebs-upgrade-jiva/busybox_app.yaml]: ok

File: litmus/director/openebs-upgrade-jiva/test.yml (hunk 2 above)
Comment (amitbhatt818) [reply litmus/director/openebs-upgrade-jiva/test.yml]: This test-case is for upgrade application that's why only the upgrade part is done from director in this PR.
For installation we have separate test cases. If it is good to have installation also director itself then i can make change in PR @AmitKumarDas 
cc: @harshshekhar15

File: litmus/director/openebs-upgrade-jiva/busybox_app.yaml (hunk 1 above)
Comment (amitbhatt818) [reply litmus/director/openebs-upgrade-jiva/busybox_app.yaml]: Changed to mongo statefulset

File: litmus/director/openebs-upgrade-jiva/test.yml (hunk 2 above)
Comment (harshshekhar15) [reply litmus/director/openebs-upgrade-jiva/test.yml]: Yes @AmitKumarDas that should be the case. We should not deploy/install any application in the test itself.

**Pull request merged**